src/services/blogs.js

```diff
@@ -1,3 +1,29 @@
-// add code here
+import { validationResult } from "express-validator";
+import { blogModel } from "../models/blog";
 
-export default {}
+const getBlogById = async (req, res) => {
+  try {
+    const validationResults = validationResult(req);
+    if (validationResults.isEmpty()) {
+      const idOfBlog = req.params.id;
+      const blog = await blogModel.findById(idOfBlog).exec();
+
+      if (blog == null) {
```

Comment:
multiple nested if/else statements are difficult to read. Learn not to nest the code unnecessarily

```
if (!validationResults.isEmpty()) {
    // some code
    return;
}

// some code for retrieving a blog
 if (!blog) {
    // some code
    return;
}

res.status(200).send(blog);
```